diff --git a/Makefile b/Makefile
index c1626b319..dfff160a0 100644
--- a/Makefile
+++ b/Makefile
@@ -215,7 +215,7 @@ endif
 
 # We trace during link so that we can determine if a libary changes under us in
 # order to re-link.  This *may* be gcc specific, so may need tweaks in future.
-LDFLAGS += -Wl,--trace
+#LDFLAGS += -Wl,--trace
 
 # ----------------------------------------------------------------------------
 # General make rules
diff --git a/frontends/Makefile.hts b/frontends/Makefile.hts
index b5af240f1..6f8aa6536 100644
--- a/frontends/Makefile.hts
+++ b/frontends/Makefile.hts
@@ -114,7 +114,7 @@ else
 endif
 
 # valid values for the TARGET
-VLDTARGET := amiga atari beos framebuffer gtk monkey riscos windows
+VLDTARGET := amiga atari beos framebuffer gtk monkey riscos windows awtk
 
 # Check for valid TARGET
 ifeq ($(filter $(VLDTARGET),$(TARGET)),)
